streams/src/test/java/org/apache/kafka/streams/integration/TaskMetadataIntegrationTest.java

```diff
@@ -77,15 +77,16 @@ public static void closeCluster() {
     private String inputTopic;
     private static StreamsBuilder builder;
     private static Properties properties;
-    private static String appId = "TaskMetadataTest_";
+    private static String appIdPrefix = "TaskMetadataTest_";
+    private static String appId;
     private AtomicBoolean process;
     private AtomicBoolean commit;
 
     @SuppressWarnings("deprecation") // Old PAPI. Needs to be migrated.
     @Before
     public void setup() {
         final String testId = safeUniqueTestName(getClass(), testName);
-        appId = appId + testId;
```

Comment:
This increases the app ID recursively at each consecutive execution of a test method of this test class.